openmicroscopy/openmicroscopy
# CLI error report

In this PR:
 * Remove usage of commons-http
 * Modify submission of files: support for both http and https.
 * Fix submission of log file (server import log) when using --logs
 * If the file that failed to import the log file (if available) is also submitted.
 * A zip is created if number of files to submit > 1.

We use https://qa.staging.openmicroscopy.org to test this PR
Test 1
 * Try to import  a faulty image e.g. squig/data_repo/test_images_bad/multi-image-pixels.ome.tif . 
 * Test logs report i.e. --logs . Check that the corresponding entry has a .log file (not .zip)
```
bin/omero import -s localhost -u root -w omero -p 4064  --report --logs PATH_TO/multi-image-pixels.ome.tif
```
 * Test upload report --upload Check that the corresponding entry has .zip
```
bin/omero import -s localhost -u root -w omero -p 4064  --report --upload PATH_TO/multi-image-pixels.ome.tif
```

Test 2
 * Try to import squig/data_repo/test_images_broken/broken_images_scenario/fails_at_processing/wrong-light-source-mk3.ome
 * Test logs. Make sure that there is no NPE in console and that the error is submitted. No log file
 * Test upload. In that case, only the file should be upload since the log is not available.


The tests under blitz will have to reviewed, To be done in another PR.
